src/main/java/mods/railcraft/client/particles/ParticleFireSpark.java

```diff
@@ -100,6 +101,16 @@ public void onUpdate() {
         }
         this.particleAge++;
 
+        /**
+         * Called to spawn smoke particles with the entity
+         */
+        float f = (float)this.particleAge / (float)this.particleMaxAge;
+
+        if (this.rand.nextFloat() > f)
+        {
+            this.worldObj.spawnParticle(EnumParticleTypes.SMOKE_NORMAL, this.posX, this.posY, this.posZ, this.motionX, this.motionY, this.motionZ, 0);
+        }
```

Comment:
I did think this was a little strange, I'll need to check what it does to the recharge animation too.